### PR TITLE
🐛 [google-analytics-data-api] [tiktok-marketing] bump CDK version and fix state updater if no cursor_field

### PR DESCRIPTION
**Pull request by brianjlai**
I was doing some inspection of recent CAT nightly build test failures and found some issues that were discovered after we moved to full refresh state messages.

The issue is that these connectors have full_refresh streams that implement an overridden `get_updated_state()` method. However, in being full refresh, they don't define a `cursor_field` so this make their usage of this method not work properly. For these connectors, we should not attempt to extract the state value from the latest record because we don't have a cursor field to do so.

Note:
In addition to fixing this, I will also make a change at the CDK level to prevent this for other connectors. However, this is really a flaw of the connector's implementation so I want to fix this here too.

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migration_cohortspec/test_config.json [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {
-  "credentials": {
-    "auth_type": "Service",
-    "credentials_json": ""
-  },
+  "credentials": { "auth_type": "Service", "credentials_json": "" },
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migration_cohortspec/test_config.json:2]: this was only modified when I ran formatting

File: airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migrations/test_config.json [hunk 2]
```diff
@@ -1,8 +1,5 @@
 {
-  "credentials": {
-    "auth_type": "Service",
-    "credentials_json": ""
-  },
+  "credentials": { "auth_type": "Service", "credentials_json": "" },
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migrations/test_config.json:2]: also only modified after running formatting

File: airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migration_cohortspec/test_config_migration_cohortspec.py [hunk 3]
```diff
@@ -32,7 +32,8 @@ def test_migrate_config(capsys):
     # migrate the test_config
     migration_instance.migrate(SOURCE_INPUT_ARGS, SOURCE)
 
-    control_msg = json.loads(capsys.readouterr().out)
+    what = capsys.readouterr().out
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_migration_cohortspec/test_config_migration_cohortspec.py:35]: TY for readability cleanup.

File: airbyte-integrations/connectors/source-google-analytics-data-api/pyproject.toml [hunk 4]
```diff
@@ -19,7 +19,7 @@ include = "source_google_analytics_data_api"
 python = "^3.9,<3.12"
 cryptography = "==37.0.4"
 requests = "==2.31.0"
-airbyte-cdk = "==0.61.0"
+airbyte-cdk = "^0"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-google-analytics-data-api/pyproject.toml:22]: Thank you!

**Review by natikgadzhi — APPROVED** [review]
I might not be the biggest brain in the area, but these look good to me. :shipit:

**Pull request merged**